File: Haskell/ABC/ABC003/ABC003_A_v01.hs
-- [contest]  ABC 003
-- [task]     A
-- [URL]      https://atcoder.jp/contests/abc003/tasks/abc003_1
-- [compiler] Haskell (GHC 7.10.3)
-- [status]   https://atcoder.jp/contests/abc003/submissions/6607929 : AC

-- modules to import --

import qualified Data.ByteString.Char8
import qualified Data.Maybe
import           Prelude

-- functions for this task --

readInteger' = Prelude.fst . Data.Maybe.fromJust . Data.ByteString.Char8.readInteger
getInteger   = readInteger' <$> Data.ByteString.Char8.getLine

task_A :: Integer -> Integer -> Double
task_A num_task_total num_task
        | num_task > 0      = (1.0e+004 :: Double) / (Prelude.fromIntegral $ num_task_total) * (Prelude.fromIntegral $ num_task) + task_A (num_task_total) (num_task - 1)
        | Prelude.otherwise = 0.0e+000

-- the main process is as follows --

main :: IO ()
main = do

        -- STEP.01
        -- read out the given data
        num_task_total <- getInteger

        -- STEP.02
        -- calculate & output the answer of this task
        Prelude.print $ task_A (num_task_total) (num_task_total)

-- EOF --

File: C(GCC5.4.1)/ABC/ABC003/ABC003_A_v01.c
// AtCoder Beginner Contest 003 A
// [URL]        https://atcoder.jp/contests/abc003/tasks/abc003_1
// [compiler]   C (GCC 5.4.1)
// [submission] https://atcoder.jp/contests/abc003/submissions/4618390 : AC

#include <stdio.h>

int calc_salary (int num_tasks)
{
	return 5000 * (num_tasks + 1); // ( num_tasks * (num_tasks + 1) / 2 ) / num_tasks * 10000
}

int main (void)
{
	/* variables for main process */
	int num_tasks;

	/* support variables for main process */
	int rtvl_scanf;

	// STEP.01
	// read out the target integers
	rtvl_scanf = scanf ("%d", &num_tasks);

	// STEP.02
	// calculate & output the salary
	printf ("%d\n", calc_salary (num_tasks));

	// STEP.TRUE_END
	return 0;
}
